data/pigui/modules/saveloadgame.lua

```diff
@@ -120,18 +113,28 @@ local function closeAndLoadOrSave()
 end
 
 ui.saveLoadWindow = ModalWindow.New("LoadGame", function()
-	local mode = ui.saveLoadWindow.mode == 'SAVE' and lui.SAVE or lui.LOAD
-	optionTextButton(mode, nil, selectedSave ~= nil and selectedSave ~= '' and saveIsValid, closeAndLoadOrSave)
-	ui.sameLine()
-	optionTextButton(lui.CANCEL, nil, true, closeAndClearCache)
+	local saving = ui.saveLoadWindow.mode == "SAVE"
+	local other_height = optionButtonSize.y + ui.getItemSpacing().y * 2 + ui.getWindowPadding().y * 2
+	if saving then
+		other_height = other_height + ui.getFrameHeightWithSpacing()
+	end
+	ui.child("savefiles", Vector2(-1, winSize.y - other_height), function()
+		showSaveFiles()
+	end)
+
+	ui.separator()
 
-	if ui.saveLoadWindow.mode == "SAVE" then
+	if saving then
 		selectedSave = ui.inputText("##saveFileName", selectedSave or "", {})
 	end
```

Comment:
This text box should be on the same line as the SAVE / CANCEL buttons; you can probably calculate the button size or just rearrange things so that this is drawn after the buttons. It's extremely unlikely / impossible for the user to input a character *and* click Save on the same frame, so I'd say there's no data ordering issues from having this later in the function.

Alternatively, if that's too big of a change to make right now, please use `ui.nextItemWidth(-1.0)` so it spans the width of the text box.

Reply:
I think I'll align the buttons on the right, and calculate the correct size for the textbox.